recipes/c-blosc2/all/conanfile.py

```diff
@@ -55,78 +47,99 @@ def config_options(self):
+            self.options["zlib-ng"].zlib_compat = False
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
         if self.options.with_lz4:
             self.requires("lz4/1.9.3")
-        if self.options.with_zlib == "zlib-ng":
+        if self.options.with_zlib in ["zlib-ng", "zlib-ng-compat"]:
             self.requires("zlib-ng/2.0.6")
         elif self.options.with_zlib == "zlib":
             self.requires("zlib/1.2.12")
         if self.options.with_zstd:
             self.requires("zstd/1.5.2")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
-
-    @functools.lru_cache(1)
-    def _configure_cmake(self):
-        cmake = CMake(self)
-        cmake.definitions["BLOSC_IS_SUBPROJECT"] = False
-        cmake.definitions["BLOSC_INSTALL"] = True
-        cmake.definitions["BUILD_STATIC"] = not self.options.shared
-        cmake.definitions["BUILD_SHARED"] = self.options.shared
-        cmake.definitions["BUILD_TESTS"] = False
-        cmake.definitions["BUILD_FUZZERS"] = False
-        cmake.definitions["BUILD_BENCHMARKS"] = False
-        cmake.definitions["BUILD_EXAMPLES"] = False
+        get(self, **self.conan_data["sources"][self.version],
+                  destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.cache_variables["BLOSC_IS_SUBPROJECT"] = False
+        tc.cache_variables["BLOSC_INSTALL"] = True
+        tc.cache_variables["BUILD_STATIC"] = not bool(self.options.shared)
+        tc.cache_variables["BUILD_SHARED"] = bool(self.options.shared)
+        tc.cache_variables["BUILD_TESTS"] = False
+        tc.cache_variables["BUILD_FUZZERS"] = False
+        tc.cache_variables["BUILD_BENCHMARKS"] = False
+        tc.cache_variables["BUILD_EXAMPLES"] = False
         simd_intrinsics = self.options.get_safe("simd_intrinsics", False)
-        cmake.definitions["DEACTIVATE_AVX2"] = simd_intrinsics != "avx2"
-        cmake.definitions["DEACTIVATE_LZ4"] = not self.options.with_lz4
-        cmake.definitions["PREFER_EXTERNAL_LZ4"] = self.options.with_lz4
-        cmake.definitions["DEACTIVATE_ZLIB"] = self.options.with_zlib == None
-        cmake.definitions["PREFER_EXTERNAL_ZLIB"] = self.options.with_zlib != None
-        cmake.definitions["DEACTIVATE_ZSTD"] = not self.options.with_zstd
-        cmake.definitions["PREFER_EXTERNAL_ZSTD"] = self.options.with_zstd
-        cmake.definitions["BUILD_PLUGINS"] = self.options.with_plugins
-        cmake.configure(build_folder=self._build_subfolder)
-        return cmake
+        tc.cache_variables["DEACTIVATE_AVX2"] = simd_intrinsics != "avx2"
+        tc.cache_variables["DEACTIVATE_LZ4"] = not bool(self.options.with_lz4)
+        tc.cache_variables["PREFER_EXTERNAL_LZ4"] = True
+        tc.cache_variables["DEACTIVATE_ZLIB"] = self.options.with_zlib == None
```

Comment:
```suggestion
        tc.cache_variables["DEACTIVATE_ZLIB"] = self.options.with_zlib is None
```
https://www.educative.io/answers/what-is-the-none-keyword-in-python